src/middlewared/middlewared/plugins/reporting.py

```diff
@@ -1060,30 +1065,75 @@ def run(self):
 
             data['cpu'] = {}
             # Get CPU usage %
-            # cp_times has values for all cores
-            cp_times = sysctl.filter('kern.cp_times')[0].value
-            # cp_time is the sum of all cores
-            cp_time = sysctl.filter('kern.cp_time')[0].value
-            if cp_times_last:
+            if osc.IS_FREEBSD:
+                num_times = 5
+                # cp_times has values for all cores
+                cp_times = sysctl.filter('kern.cp_times')[0].value
+                # cp_time is the sum of all cores
+                cp_time = sysctl.filter('kern.cp_time')[0].value
+            elif osc.IS_LINUX:
+                num_times = 10
+                with open('/proc/stat') as f:
+                    stat = f.read()
+                cp_times = []
+                cp_time = []
+                for line in stat.split('\n'):
+                    if line.startswith('cpu'):
+                        line_ints = [int(i) for i in line[5:].strip().split()]
+                        # cpu has a sum of all cpus
+                        if line[3] == ' ':
+                            cp_time = line_ints
+                        # cpuX is for each core
+                        else:
+                            cp_times += line_ints
+                    else:
+                        break
+            else:
+                cp_time = cp_times = None
+
+            if cp_time and cp_times and cp_times_last:
                 # Get the difference of times between the last check and the current one
                 # cp_time has a list with user, nice, system, interrupt and idle
                 cp_diff = list(map(lambda x: x[0] - x[1], zip(cp_times, cp_times_last)))
-                cp_nums = int(len(cp_times) / 5)
+                cp_nums = int(len(cp_times) / num_times)
                 for i in range(cp_nums):
-                    data['cpu'][i] = self.get_cpu_usages(cp_diff[i * 5:i * 5 + 5])
+                    data['cpu'][i] = self.get_cpu_usages(cp_diff[i * num_times:i * num_times + num_times])
 
                 cp_diff = list(map(lambda x: x[0] - x[1], zip(cp_time, cp_time_last)))
                 data['cpu']['average'] = self.get_cpu_usages(cp_diff)
+
             cp_time_last = cp_time
             cp_times_last = cp_times
 
             # CPU temperature
             data['cpu']['temperature'] = {}
-            for i in itertools.count():
-                v = sysctl.filter(f'dev.cpu.{i}.temperature')
-                if not v:
-                    break
-                data['cpu']['temperature'][i] = v[0].value
+            if osc.IS_FREEBSD:
+                for i in itertools.count():
+                    v = sysctl.filter(f'dev.cpu.{i}.temperature')
+                    if not v:
+                        break
+                    data['cpu']['temperature'][i] = v[0].value
+            elif osc.IS_LINUX:
+                cp = subprocess.run(['sensors', '-j'], capture_output=True, text=True)
```

Comment:
Can this be worth to use https://pypi.org/project/PySensors/ to avoid launching new process every time in a loop?

Reply:
Sure, I can create you a ticket to create a package for that and change usage